chirp/drivers/kguv980p.py

```diff
@@ -0,0 +1,2583 @@
+        #                   RadioSettingValueMap(MUTE_MODE_MAP,
+        #                   _mem.mute_mode))
+        # mem.extra.append(rs)
+
+        return mem
+
+    def _set_tone(self, mem, _mem):
+        def _set_dcs(code, pol):
+            # MRT Change to 0x8000 to
+            # set the bit for DCS- code is a decimal version
+            # of the code # - NOT OCTAL
+            val = int("%i" % code, 10) | 0x8000
+            if pol == "R":
+                # MRT Change to 0x4000 to set the bit for
+                # i/R polarity
+                val += 0x4000
+            return val
+
+        rx_mode = tx_mode = None
+        rxtone = txtone = 0x0000
+
+        if mem.tmode == "Tone":
+            tx_mode = "Tone"
+            rx_mode = None
+            txtone = int(mem.rtone * 10)
+        elif mem.tmode == "TSQL":
+            rx_mode = tx_mode = "Tone"
+            rxtone = txtone = int(mem.ctone * 10)
+        elif mem.tmode == "DTCS":
+            tx_mode = rx_mode = "DTCS"
+            txtone = _set_dcs(mem.dtcs, mem.dtcs_polarity[0])
+            rxtone = _set_dcs(mem.dtcs, mem.dtcs_polarity[1])
+        elif mem.tmode == "Cross":
+            tx_mode, rx_mode = mem.cross_mode.split("->")
+            if tx_mode == "DTCS":
+                txtone = _set_dcs(mem.dtcs, mem.dtcs_polarity[0])
+            elif tx_mode == "Tone":
+                txtone = int(mem.rtone * 10)
+            if rx_mode == "DTCS":
+                rxtone = _set_dcs(mem.rx_dtcs, mem.dtcs_polarity[1])
+            elif rx_mode == "Tone":
+                rxtone = int(mem.ctone * 10)
+
+        _mem.rxtone = rxtone
+        _mem.txtone = txtone
+
+        LOG.debug("Set TX %s (%i) RX %s (%i)" %
+                  (tx_mode, _mem.txtone, rx_mode, _mem.rxtone))
+
+    def set_memory(self, mem):
+        # _mem = Stored Memory value
+        # mem = New value from user entry
+        number = mem.number
+        _mem = self._memobj.memory[number]
+        _nam = self._memobj.names[number]
+        _valid = self._memobj.valid[mem.number]
+
+        # if mem.empty:
+        #     _mem.set_raw("\x00" * (_mem.size() / 8))
+        #     self._memobj.valid[number] = 0
+        #     self._memobj.names[number].set_raw("\x00" * (_nam.size() / 8))
+        #     return
+
+        if mem.empty:
+            LOG.debug("Mem %s is Empty", number)
+            self._memobj.valid[number] = 0xFF
+            LOG.debug("Set Mem %s Not Valid", number)
+            _mem.rxfreq = 0xFFFFFFFF
+            LOG.debug("Set Rx Freq = FFFFFFF")
+            _mem.txfreq = 0xFFFFFFFF
+            LOG.debug("Set Tx Freq = FFFFFFF")
+            self._memobj.names[number].set_raw("\xFF" * (_nam.size() / 8))
+            LOG.debug("Name %s Cleared", number)
+            # The following line is a hack to make CPS happy and not
+            # show memory entries that were deleted with CHIRP
+            # _mem=0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
```

Comment:
This concerns me. Does the code in its current form confuse the CPS or not? If it does, we need to fix that before we let people use this. If not, then this should be removed.

If you want to set the entire _mem structure to 0xFF bytes, use _mem.set_raw() (which is common practice when erasing memories on radios that want that).

Reply:
these are comments that can be deleted

Reply:
Okay cool :)